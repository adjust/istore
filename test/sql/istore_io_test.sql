BEGIN;
-- istore_io should persist istores;
-- ./spec/istore_io_spec.rb:8;
CREATE EXTENSION istore;
CREATE TABLE istore_io AS SELECT '1=>1,1=>2'::istore;
SELECT * FROM istore_io;
ROLLBACK;
BEGIN;
-- istore_io should persist istores;
-- ./spec/istore_io_spec.rb:14;
CREATE EXTENSION istore;
CREATE TABLE istore_io AS SELECT '"1"=>"1", "1"=>"2"'::istore;
SELECT * FROM istore_io;
ROLLBACK;
BEGIN;
-- istore_io should persist istores;
-- ./spec/istore_io_spec.rb:20;
CREATE EXTENSION istore;
CREATE TABLE istore_io AS SELECT '"1"=>"1", "-1"=>"2"'::istore;
SELECT * FROM istore_io;
ROLLBACK;
BEGIN;
-- istore_io should persist istores;
-- ./spec/istore_io_spec.rb:26;
CREATE EXTENSION istore;
CREATE TABLE istore_io AS SELECT '"-1"=>"+1","1"=>"2"'::istore;
SELECT * FROM istore_io;
ROLLBACK;
BEGIN;
-- istore_io should persist istores;
-- ./spec/istore_io_spec.rb:32;
CREATE EXTENSION istore;
CREATE TABLE istore_io AS SELECT ' "-1"=>"+1","1"=>"2"'::istore;
SELECT * FROM istore_io;
ROLLBACK;
BEGIN;
-- istore_io should persist istores;
-- ./spec/istore_io_spec.rb:38;
CREATE EXTENSION istore;
CREATE TABLE istore_io AS SELECT '"-1"=> "+1","1"=>"2"'::istore;
SELECT * FROM istore_io;
ROLLBACK;
BEGIN;
-- istore_io should persist istores;
-- ./spec/istore_io_spec.rb:44;
CREATE EXTENSION istore;
CREATE TABLE istore_io AS SELECT '"-1"=>"+1", "1"=>"2"'::istore;
SELECT * FROM istore_io;
ROLLBACK;
BEGIN;
-- istore_io should persist istores;
-- ./spec/istore_io_spec.rb:50;
CREATE EXTENSION istore;
CREATE TABLE istore_io AS SELECT '"-1"=>"+1","1"=>"2" '::istore;
SELECT * FROM istore_io;
ROLLBACK;
BEGIN;
-- istore_io should persist istores;
-- ./spec/istore_io_spec.rb:56;
CREATE EXTENSION istore;
CREATE TABLE istore_io AS SELECT '-1=>+1,1=>2 '::istore;
SELECT * FROM istore_io;
ROLLBACK;
BEGIN;
-- istore_io should persist istores;
-- ./spec/istore_io_spec.rb:62;
CREATE EXTENSION istore;
CREATE TABLE istore_io AS SELECT '"-1"=>"+1" ,"1"=>"2"'::istore;
SELECT * FROM istore_io;
ROLLBACK;
