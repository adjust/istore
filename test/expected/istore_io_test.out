BEGIN;
CREATE EXTENSION istore;
CREATE TABLE istore_io AS SELECT '-2147483647 => 10, -10 => -9223372036854775807, 0 => 5, 10 => 9223372036854775806, 2147483647 => 10'::bigistore;
SELECT * FROM istore_io;
                                                   bigistore                                                   
---------------------------------------------------------------------------------------------------------------
 "-2147483647"=>"10", "-10"=>"-9223372036854775807", "0"=>"5", "10"=>"9223372036854775806", "2147483647"=>"10"
(1 row)

ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
CREATE TABLE istore_io AS SELECT '"-1"=>"+1","1"=>"2"'::bigistore;
SELECT * FROM istore_io;
      bigistore      
---------------------
 "-1"=>"1", "1"=>"2"
(1 row)

ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
CREATE TABLE istore_io AS SELECT ' "-1"=>"+1","1"=>"2"'::bigistore;
SELECT * FROM istore_io;
      bigistore      
---------------------
 "-1"=>"1", "1"=>"2"
(1 row)

ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
CREATE TABLE istore_io AS SELECT ''::bigistore;
SELECT * FROM istore_io;
 bigistore 
-----------
 
(1 row)

ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT istore_to_json('-2147483647 => 10, -10 => -9223372036854775807, 0 => 5, 10 => 9223372036854775806, 2147483647 => 10'::bigistore);
                                            istore_to_json                                             
-------------------------------------------------------------------------------------------------------
 {"-2147483647": 10, "-10": -9223372036854775807, "0": 5, "10": 9223372036854775806, "2147483647": 10}
(1 row)

ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '2=>4, 1=>foo, 5=>17'::bigistore;
ERROR:  invalid input syntax for istore: "2=>4, 1=>foo, 5=>17"
LINE 1: SELECT '2=>4, 1=>foo, 5=>17'::bigistore;
               ^
DETAIL:  unexpected sign f, in istore value
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '2=>4, 1=>5foo, 5=>17'::bigistore;
ERROR:  invalid input syntax for istore: "2=>4, 1=>5foo, 5=>17"
LINE 1: SELECT '2=>4, 1=>5foo, 5=>17'::bigistore;
               ^
DETAIL:  unexpected sign f, in istore value
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '2=>4, 1=>18446744073709551612, 5=>17'::bigistore;
ERROR:  istore "2=>4, 1=>18446744073709551612, 5=>17" is out of range
LINE 1: SELECT '2=>4, 1=>18446744073709551612, 5=>17'::bigistore;
               ^
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '2=>4, 1=>-18446744073709551614, 5=>17'::bigistore;
ERROR:  istore "2=>4, 1=>-18446744073709551614, 5=>17" is out of range
LINE 1: SELECT '2=>4, 1=>-18446744073709551614, 5=>17'::bigistore;
               ^
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '2=>4, 54foo=>5, 5=>17'::bigistore;
ERROR:  invalid input syntax for istore: "2=>4, 54foo=>5, 5=>17"
LINE 1: SELECT '2=>4, 54foo=>5, 5=>17'::bigistore;
               ^
DETAIL:  unexpected sign f, in istore key
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '2=>4, foo=>5, 5=>17'::bigistore;
ERROR:  invalid input syntax for istore: "2=>4, foo=>5, 5=>17"
LINE 1: SELECT '2=>4, foo=>5, 5=>17'::bigistore;
               ^
DETAIL:  unexpected sign f, in istore key
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '2=>4, 4000000000=>5, 5=>17'::bigistore;
ERROR:  istore "2=>4, 4000000000=>5, 5=>17" is out of range
LINE 1: SELECT '2=>4, 4000000000=>5, 5=>17'::bigistore;
               ^
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '2=>4, -4000000000=>5, 5=>17'::bigistore;
ERROR:  istore "2=>4, -4000000000=>5, 5=>17" is out of range
LINE 1: SELECT '2=>4, -4000000000=>5, 5=>17'::bigistore;
               ^
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '2=>4, 10=5, 5=>17'::bigistore;
ERROR:  invalid input syntax for istore: "2=>4, 10=5, 5=>17"
LINE 1: SELECT '2=>4, 10=5, 5=>17'::bigistore;
               ^
DETAIL:  unexpected sign 5, expected '>'
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
CREATE TABLE istore_io AS SELECT '-2147483647 => 10, -10 => -2147483647, 0 => 5, 10 => 2147483647, 2147483647 => 10'::istore;
SELECT * FROM istore_io;
                                           istore                                            
---------------------------------------------------------------------------------------------
 "-2147483647"=>"10", "-10"=>"-2147483647", "0"=>"5", "10"=>"2147483647", "2147483647"=>"10"
(1 row)

ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
CREATE TABLE istore_io AS SELECT '"-1"=>"+1","1"=>"2"'::istore;
SELECT * FROM istore_io;
       istore        
---------------------
 "-1"=>"1", "1"=>"2"
(1 row)

ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
CREATE TABLE istore_io AS SELECT ' "-1"=>"+1","1"=>"2"'::istore;
SELECT * FROM istore_io;
       istore        
---------------------
 "-1"=>"1", "1"=>"2"
(1 row)

ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
CREATE TABLE istore_io AS SELECT ''::istore;
SELECT * FROM istore_io;
 istore 
--------
 
(1 row)

ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT istore_to_json('-2147483647 => 10, -10 => -2147483647, 0 => 5, 10 => 2147483647, 2147483647 => 10'::istore);
                                   istore_to_json                                    
-------------------------------------------------------------------------------------
 {"-2147483647": 10, "-10": -2147483647, "0": 5, "10": 2147483647, "2147483647": 10}
(1 row)

ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '2=>4, 1=>foo, 5=>17'::istore;
ERROR:  invalid input syntax for istore: "2=>4, 1=>foo, 5=>17"
LINE 1: SELECT '2=>4, 1=>foo, 5=>17'::istore;
               ^
DETAIL:  unexpected sign f, in istore value
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '2=>4, 1=>5foo, 5=>17'::istore;
ERROR:  invalid input syntax for istore: "2=>4, 1=>5foo, 5=>17"
LINE 1: SELECT '2=>4, 1=>5foo, 5=>17'::istore;
               ^
DETAIL:  unexpected sign f, in istore value
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '2=>4, 1=>4294967294, 5=>17'::istore;
ERROR:  integer out of range
LINE 1: SELECT '2=>4, 1=>4294967294, 5=>17'::istore;
               ^
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '2=>4, 1=>-4294967294, 5=>17'::istore;
ERROR:  integer out of range
LINE 1: SELECT '2=>4, 1=>-4294967294, 5=>17'::istore;
               ^
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '2=>4, 54foo=>5, 5=>17'::istore;
ERROR:  invalid input syntax for istore: "2=>4, 54foo=>5, 5=>17"
LINE 1: SELECT '2=>4, 54foo=>5, 5=>17'::istore;
               ^
DETAIL:  unexpected sign f, in istore key
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '2=>4, foo=>5, 5=>17'::istore;
ERROR:  invalid input syntax for istore: "2=>4, foo=>5, 5=>17"
LINE 1: SELECT '2=>4, foo=>5, 5=>17'::istore;
               ^
DETAIL:  unexpected sign f, in istore key
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '2=>4, 4000000000=>5, 5=>17'::istore;
ERROR:  istore "2=>4, 4000000000=>5, 5=>17" is out of range
LINE 1: SELECT '2=>4, 4000000000=>5, 5=>17'::istore;
               ^
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '2=>4, -4000000000=>5, 5=>17'::istore;
ERROR:  istore "2=>4, -4000000000=>5, 5=>17" is out of range
LINE 1: SELECT '2=>4, -4000000000=>5, 5=>17'::istore;
               ^
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '2=>4, 10=5, 5=>17'::istore;
ERROR:  invalid input syntax for istore: "2=>4, 10=5, 5=>17"
LINE 1: SELECT '2=>4, 10=5, 5=>17'::istore;
               ^
DETAIL:  unexpected sign 5, expected '>'
ROLLBACK;
